tests/namespaced.rs

```diff
@@ -0,0 +1,357 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::net::{IpAddr, SocketAddr};
+
+use hyper::{Body, Method};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpStream;
+use tracing::{error, info};
+
+use ztunnel::identity;
+use ztunnel::identity::CertificateProvider;
+use ztunnel::test_helpers::components::WorkloadManager;
+use ztunnel::test_helpers::helpers::initialize_telemetry;
+use ztunnel::test_helpers::netns::{Namespace, Resolver};
+use ztunnel::test_helpers::*;
+
+macro_rules! function {
+    () => {{
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        let name = type_name_of(f);
+        &name[..name.len() - 3]
+    }};
+}
+
+/// setup_netns_test prepares a test using network namespaces. This checks we have root,
+/// and automatically setups up a namespace based on the test name (to avoid conflicts).
+macro_rules! setup_netns_test {
```

Comment:
why does this have to be a macro?

Reply:
It was intended to do more... but it does `return` which is a bit nice